Frameworks/UIKit/UIScrollView.mm

```diff
@@ -139,7 +145,42 @@ @implementation UIScrollView {
     WXCScrollBarVisibility _previousVerticalScrollBarVisibility;
 }
 
+- (id)_dManipGestureCallback:(_UIDMPanGestureRecognizer*)gesture {
+    if (DEBUG_DMANIP_GESTURE) {
+        TraceVerbose(TAG, L"_dManipGestureCallback fired");
+    }
+
+    // check the gesture state, if it is already being cancelled, don't start direct manipulation
+    if (gesture.state == UIGestureRecognizerStateCancelled) {
+        if (DEBUG_DMANIP_GESTURE) {
+            TraceVerbose(TAG, L"dManipGesture cancelled");
+        }
+
+        return self;
+    }
+
+    // setting manipulationMode of scrollview (on its canvas) chain to be System.
+    // so that direct manipulation will be effective on all scrollviews
+    [self _setManipulationMode:WUXIManipulationModesSystem recursive:YES];
+
+    const std::vector<TouchInfo>& pointers = [gesture _getTouches];
+    for (size_t i = 0; i < pointers.size(); ++i) {
+        UITouch* touch = pointers[i].touch;
+        // start Direct manipuation only if active touch presents
+        if (touch.phase != UITouchPhaseCancelled && touch->_routedEventArgs.pointer.pointerDeviceType == WDIPointerDeviceTypeTouch) {
```

Comment:
yes

---
In reply to: [86442520](https://github.com/Microsoft/WinObjC/pull/1305#discussion_r86442520) [](ancestors = 86442520)